### DISCUSSION TITLE
Lazy Loading

### DISCUSSION THREAD

**Discussion by sp9324** [meteor/meteor#13405] [Feature Requests]
I would like to implement Lazy Loading in the webapp. This would enhance user experience and also boost performance.

**Comment by StorytellerCZ** [comment]
I do believe that depends on your front-end if I'm not mistaken.
There is this old article for `react-loadable`: https://medium.com/official-meteor-blog/meteor-1-5-react-loadable-f029a320e59c
But if you are using React, you can use `React.lazy` without a problem.

**Comment by sp9324** [comment]
> I do believe that depends on your front-end if I'm not mistaken. There is this old article for `react-loadable`: https://medium.com/official-meteor-blog/meteor-1-5-react-loadable-f029a320e59c But if you are using React, you can use `React.lazy` without a problem.

I meant to comment this as an issue for Hacktoberfest, actually. (Which is why I started this conversation in Feature Requests). I would like to implement Lazy Loading in the website. Thanks.
Please assign :)

**Reply by StorytellerCZ** [reply]
I don't follow. Which website?

**Comment by jeffreyflynt** [comment]
If you are using React and react-router-dom@v6 here is a guide that I use in my Meteor React apps that you can follow: https://remix.run/blog/lazy-loading-routes

**Comment by banjerluke** [comment]
For images, I use the [vanilla-lazyload](https://github.com/verlok/vanilla-lazyload) npm package, which works great and is platform-agnostic. These days, however, you probably don't need a library for this can just [use the `loading="lazy"` attribute on your `<img>` tag](https://web.dev/articles/browser-level-image-lazy-loading) (supported by all modern browsers and degrades gracefully).

I realize now that @sp9324 is probably talking about lazy-loading chunks of JS code, or UI components, etc. and not images per-se, but I'll leave this comment in case it's useful to anyone.
